src/coffea/jetmet_tools/CorrectedMETFactory.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, name_map):
 
         self.name_map = name_map
 
-    def build(self, in_MET, in_corrected_jets):
+    def build(self, in_MET, type1_MET, in_corrected_jets):
```

Comment:
Couldn't you just set the Type1 MET as the MET that is input to the build function? This seems odd.

Reply:
- In order to obtain the corresponding uncertainty values of MET after correction for `JEC `and `JER`, we must use `RawMET` as input for correction.RawMET means uncorrected PF MET, while MET refers to Type-1 corrected PF MET. If using MET as input, it means that it has been corrected twice.

- But if we want to obtain the `UnclusterdEnergy` uncertainty value of MET, you must use MET as input, as in `NanoAOD`, only MET has the branch for `MetUnclusterEnUpDeltaX/Y`, while RawMET does not. Therefore, it is necessary to use MET to obtain the `UnclusteredEnergy` uncertainty value.

So I have to input both `in_MET` and `type1_MET` meanwhile, they represent RawMET and MET respectively, for example:
`met  = self._jmeu.corrected_met(event.RawMET,event.MET, jets_col, event.fixedGridRhoFastjetAll, event.caches[0])`

Reply:
Perhaps a better way to put this is I think this can be solved by changing the flexibility in the configuration as opposed to changing the meaning of the physics objects that people get out from correctors.

Reply:
But it is obvious that the `CorrectedMETFactory` function here cannot implement the correction of `RawMET` as input. Because `RawMET` does not have the `MetUnclusterEnUpDeltaX/Y` branch, a bug will occur. How can I modify the configuration file to solve this problem? At the same time, I also need to obtain the uncertainty of `MET_UnclusterdEnergy_up/down`

Reply:
This would be better solved by making a new met object with the appropriate substitutions.
```python3
raw_met = events.RawMET
met_to_correct = events.MET
met_to_correct["pt"] = raw_met.pt
met_to_correct["phi"] = raw_met.phi

# then go on to correct met_to_correct, adjusting configuration as appropriate
```

Reply:
I know what you mean! Great! 
This will not spoils the validity of the systemic variations,  the uncertainty of `MET_UnclusteredEnergy` is calculated using the `corrected MET_pt` value through `corrected_polar_met`, rather than RawMET.pt. 
You are incredibly smart, thank you so much for your patient and thoughtful answers.